src/detect-engine-address-ipv6.c

```diff
@@ -1659,48 +1392,30 @@ static int AddressTestIPv6CutNot02(void)
     DetectAddress *b = NULL;
     DetectAddress *temp = NULL;
     struct in6_addr in6;
-    int result = 1;
 
-    if ( (a = DetectAddressInit()) == NULL)
-        goto error;
-    if ( (temp = DetectAddressInit()) == NULL)
-        goto error;
+    a = DetectAddressInit();
```

Comment:
declare `a` here

Similar for the rest of the patch: please declare vars on first use for more compactness